cmd/kubeadm/app/phases/upgrade/postupgrade_test.go

```diff
@@ -169,6 +175,144 @@ func TestWriteKubeletConfigFiles(t *testing.T) {
 	}
 }
 
+func TestUnupgradedControlPlaneInstances(t *testing.T) {
+	testCases := []struct {
+		name          string
+		pods          []corev1.Pod
+		currentNode   string
+		expectedNodes []string
+		expectError   bool
+	}{
+		{
+			name: "one unupgraded instance",
+			pods: []corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-1",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-1",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v1"},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-2",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-2",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v2"},
```

Comment:
let's use v2 for nodes that are already upgraded and v1 for nodes that need upgrade.
currently it's in reverse v2 == needs upgrade, which is a bit confusing.

Reply:
fixed